### PR TITLE
Respect `fallback` also without `json` flag in `getValue`

### LINKED ISSUE #212: Support `getValue` fallback without `json` flag
The `fallback` option of `getValue` is currently being ignored without the `json` flag being set.
I'm not seeing any reason for not supporting `fallback` in this case but I might be missing something.

I'm open to contribute this change also myself if a maintainer agrees.

### PR DESCRIPTION
**Pull request by levrik**
Allows the following usage:

```ts
flagsmith.getValue('refresh_interval', { fallback: 10000 });
```

Fixes #212

### CODE REVIEW

File: flagsmith-core.ts [hunk 1]
```diff
@@ -818,15 +818,19 @@ const Flagsmith = class {
...
+            return options.fallback;
+        }
+
         if (options?.json) {
             try {
                 if (res === null) {
-                    this.log("Tried to parse null flag as JSON: " + key)
-                    return options.fallback;
+                    this.log("Tried to parse null flag as JSON: " + key);
+                    return undefined;
```
Comment (levrik) [inline flagsmith-core.ts:829]: I changed this to `undefined` as we can only end up with an `undefined` fallback here anyway.
Returning `options.fallback` seemed a bit confusing.

Comment (kyle-ssg) [reply flagsmith-core.ts:829]: That makes sense, though I think returning null might be slightly less confusing right ?

**Review by kyle-ssg — APPROVED** [review]

**Comment by kyle-ssg** [comment]
Thanks! This is now in 3.22.1 of flagsmith,flagsmith-es and react-native-flagsmith

File: flagsmith-core.ts (hunk 1 above)
Comment (levrik) [reply flagsmith-core.ts]: I just tried to match previous behavior which would have been `undefined` as it simply returned `options.fallback`.

**Pull request merged**